### ISSUE TITLE
[`no-parsing-error`] rule should not throw for template type assertions

### ISSUE THREAD

**Issue by qnp** [vuejs/eslint-plugin-vue#2082]
**What rule do you want to change?**

`no-parsing-error`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer *errors*.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior.

**Please provide some example code that this change will affect:**

Any type assertion in templates. Here is a very naïve example:

```vue
<script setup lang="ts">
let x: string | number = 1
</script>

<template>
  {{ (x as number).toFixed(2) }}
</template>
```

**What does the rule currently do for this code?**

`Parsing error: Unexpected token as`

**What will the rule do after it's changed?**

No error.

**Additional context**

As stated by the [official VueJS documentation](https://vuejs.org/guide/typescript/overview.html#typescript-in-templates), from which I took the above example, this syntax is valid.

**Comment by FloEdelmann** [comment]
Are you using Vue 2 or Vue 3?

TypeScript syntax in template expressions is not allowed in Vue 2: https://v2.vuejs.org/v2/guide/migration-vue-2-7.html#Behavior-Differences-from-Vue-3

**Comment by FloEdelmann** [comment]
It should work in Vue 3. But since there is no response, I'm closing this. Feel free to comment new information or open a new issue.

**Comment by joris-gallot** [comment]
I confirm that I reproduce the same problem with vue3

**Comment by joris-gallot** [comment]
sorry it works with `@typescript-eslint/parser` as parser for vue template 👍
